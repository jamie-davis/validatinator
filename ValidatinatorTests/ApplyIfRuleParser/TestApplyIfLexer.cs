using System.Linq;
using TestConsole.OutputFormatting;
using TestConsoleLib;
using TestConsoleLib.Testing;
using Validatinator.ApplyIfRuleParser;
using Xunit;

namespace ValidatinatorTests.ApplyIfRuleParser
{
    public class TestApplyIfLexer
    {
        private static readonly string[] TestStrings = 
        {
            "firsterror",
            "FirstError",
            "cleaninput",
            "CleanInput",
            "identifier.another",
            "@error",
            "match(entity.field)",
            "Match,entity,",
        };

        [Fact]
        public void LexerExtractsNameTokens()
        {
            //Arrange
            var output = new Output();

            //Act
            var results = TestStrings
                .Select(s => new { String = s, Result = ApplyIfLexer.Analyse(s).ToList() })
                .ToList();

            //Assert
            var report = results.AsReport(rep => rep
                .AddColumn(r => r.String, cc => cc.Heading("Input"))
                .AddChild(r => r.Result, res => res
                    .AddColumn(r => r.TokenType, cc => {})
                    .AddColumn(r => r.Text, cc => {})));
                
            output.FormatTable(report);
            output.Report.Verify();
        }
    }
}
