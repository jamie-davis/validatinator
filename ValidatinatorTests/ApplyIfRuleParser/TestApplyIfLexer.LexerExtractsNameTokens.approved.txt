Input                                                                                          
-------------------                                                                            
firsterror                                                                                     
                                                                                               
      Token Type Text                                                                          
    ------------ ----------                                                                    
      FirstError firsterror                                                                    
                                                                                               
Input                                                                                          
-------------------                                                                            
FirstError                                                                                     
                                                                                               
      Token Type Text                                                                          
    ------------ ----------                                                                    
      FirstError firsterror                                                                    
                                                                                               
Input                                                                                          
-------------------                                                                            
cleaninput                                                                                     
                                                                                               
      Token Type Text                                                                          
    ------------ ----------                                                                    
      CleanInput cleaninput                                                                    
                                                                                               
Input                                                                                          
-------------------                                                                            
CleanInput                                                                                     
                                                                                               
      Token Type Text                                                                          
    ------------ ----------                                                                    
      CleanInput cleaninput                                                                    
                                                                                               
Input                                                                                          
-------------------                                                                            
identifier.another                                                                             
                                                                                               
      Token Type Text                                                                          
    ------------ ----------                                                                    
      Identifier identifier                                                                    
          Period .                                                                             
      Identifier another                                                                       
                                                                                               
Input                                                                                          
-------------------                                                                            
@error                                                                                         
                                                                                               
      Token Type Text                                                                          
    ------------ ------                                                                        
      Identifier @error                                                                        
                                                                                               
Input                                                                                          
-------------------                                                                            
match(entity.field)                                                                            
                                                                                               
      Token Type Text                                                                          
    ------------ ------                                                                        
           Match match                                                                         
       OpenParen (                                                                             
      Identifier entity                                                                        
          Period .                                                                             
      Identifier field                                                                         
      CloseParen )                                                                             
                                                                                               
Input                                                                                          
-------------------                                                                            
Match,entity,                                                                                  
                                                                                               
      Token Type Text                                                                          
    ------------ ------                                                                        
           Match match                                                                         
           Comma ,                                                                             
      Identifier entity                                                                        
           Comma ,                                                                             
                                                                                               
                                                                                               